continuous-integration/readme.md

```diff
@@ -1,3 +1,233 @@
+- [ ] **Code Style Checks**
+  - Code across an engineering team must be formatted to agreed coding standards. Such standards keep code consistent and most importantly easy for the team and customer(s) to read and refactor. Code styling consistency encourages collective ownership for project scrum teams and our partners.
+  - There are several open source code style validation tools available to choose from ([code style checks](https://github.com/checkstyle/checkstyle), [StyleCop](https://en.wikipedia.org/wiki/StyleCop)). The [Code Review section](https://github.com/microsoft/code-with-engineering-playbook/tree/master/code-reviews#language-specific-guidance) of the playbook has suggestions for linters and preferred styles for a number of languages.
+  - Code standards are maintained within a single configuration file. There should be a step in your build pipeline that asserts code in the latest commit conforms to the known style definition.
+- [ ] **Build Script Target**
+  - A single command should have the capability of building the system. This is also true for builds running on a CI server or on a developers local machine.
+- [ ] **No IDE Dependencies**
+  - It's essential to have a master build that's runnable through standalone scripts and not dependent on a particular IDE. Build pipeline targets can be triggered locally on their desktops through their IDE of choice. The build process should maintain enough flexibility to run within a CI server as well. As an example, dockerizing your build process offers this level of flexibility as VSCode and IntelliJ supports [docker plugin](https://code.visualstudio.com/docs/containers/overview) extensions.
+
+## Build Environment Dependencies
+
+- [ ] **Automated local environment setup**
+  - We encourage maintaining a consistent developer experience for all team members. There should be a central automated manifest / process that streamlines the installation and setup of any software dependencies. This way developers can replicate the same build environment locally as the one running on a CI server.
+  - Build automation scripts often require specific software packages and version pre-installed within the runtime environment of the OS. This presents some challenges as build processes typically version lock these dependencies.
+  - All developers on the team should be able to emulate the build environment from their local desktop regardless of their OS.
+  - Developers should trigger the automated environment setup task(s) through a single command.
+  - Well established software packaging tools like Docker, Maven, npm, etc should be considered when designing your build automation tool chain.
+- [ ] **Document local setup**
+  - The setup process for setting up a local build environment should be well documented and easy for developers to follow.
+  
+## Infrastructure as Code
```

Comment:
Can we add some commentary about "what is infrastructure"? Some ones that are getting easier and easier to manage through Terraform include:
 - Service connections required to deploy application and/or infrastructure
 - AZDO Variable Groups, Service Connections, Build Definitions, Repos, etc...